2-projects/1-todo-list/script.js

```diff
@@ -17,9 +56,23 @@ function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
   // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  let newItem = document.getElementById("todoInput");
+  populateTodoList([{ task: newItem.value, completed: false }]);
+  newItem.value = "";
 }
 
 // Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
-function deleteAllCompletedTodos() {
+function deleteAllCompletedTodos(event) {
   // Write your code here...
+  event.preventDefault();
+
+  let list = document.getElementById("todo-list");
+
+  let allItems = document.querySelectorAll("li");
+
+  allItems.forEach((item) => {
+    if (item.classList.contains("completed")) {
+      list.removeChild(item);
+    }
+  });
```

Comment:
This all works well. An alternative approach would be to add, delete or update items in the underlying todos list when an add, delete or complete/uncomplete event is called, and then call the populateTodoList with your updated todos.